src/delphi/train/iteration_params.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class IterationParams:
+    num_batches: int
+    num_steps: int
+    eval_iters: int
+    lr_decay_iters: int
+    tokens_per_iter: int
+
+
+def set_iteration_params(config, train_ds, validation_ds) -> IterationParams:
+    num_batches = len(train_ds) // config.batch_size
+    num_steps = num_batches // config.gradient_accumulation_steps
+    eval_iters = min(12, len(validation_ds) // config.batch_size)
```

Comment:
what is the 12

Reply:
Something that was already there that I didn't want to change without understanding it first: https://github.com/delphi-suite/delphi/pull/31/files#diff-c113425bb7a4b6c38858b09ca918bc89bd243d139cdca30ab5dd386d9690935bR94

Reply:
as far as I can tell this is a constant set by Karpathy to log at least every 12 batches.